ql/src/java/org/apache/hadoop/hive/ql/exec/util/DAGTraversal.java

```diff
@@ -42,9 +43,11 @@ public static void traverse(List<Task<? extends Serializable>> tasks, Function f
           children.addAll(task.getDependentTasks());
         }
         function.process(task);
+        numTasks++;
```

Comment:
This doesn't ignore the duplicate path. So, the tasks count could be bloated up. Need to relook.

Reply:
changed it to task tracker